src/hooks/useConnectWallet.ts

```diff
@@ -1,49 +1,201 @@
 "use client"
 
 import { useWalletSelector } from "@src/providers/WalletSelectorProvider"
+import type { SignAndSendTransactionsParams } from "@src/utils/myNearWalletAdapter"
+import { useEffect, useState } from "react"
+import { type Connector, useAccount, useConnect, useDisconnect } from "wagmi"
+import { useNearWalletActions } from "./useNearWalletActions"
+
+export enum SignInType {
+  NearWalletSelector = "near-wallet-selector",
+  WalletConnect = "wallet-connect",
+  Bitcoin = "bitcoin",
+}
 
 interface ConnectWalletAction {
-  handleSignIn: () => void
-  handleSignOut: () => Promise<void>
-  handleSwitchWallet: () => void
-  handleSwitchAccount: () => void
+  signIn: ({
+    id,
+    params,
+  }: {
+    id: SignInType
+    params: { connector: Connector } | undefined
+  }) => Promise<void>
+  signOut: ({ id }: { id: SignInType }) => Promise<void>
+  signMessage: ({
+    id,
+    message,
+  }: { id: SignInType; message: string }) => Promise<void>
+  sendTransaction: ({
+    id,
+    type,
+    transactions,
+  }: {
+    id: SignInType
+    type: SendTransactionType
+    transactions: SignAndSendTransactionsParams
+  }) => Promise<void>
+  connectors: Connector[]
```

Comment:
No,`connectors` is a list of available wallet connectors as provided by the `wagmi` configuration.
Here's an example from the `wagmi` [documentation](https://wagmi.sh/react/api/connectors).
In other words, we need this array to render wallet connection buttons, with each button bound to an action that connects to its corresponding wallet.

Reply:
I got what connectors are. But still think it is not needed in `ConnectWalletAction`. If we want to show available options, then it's easier to use `const { connectors } = useConnect()` or even get connectors directly from config.

Overall such data locality is nice, but it overloads `ConnectWalletAction` interface. After careful consideration I think `signMessage` and `sendTransaction` are looking out of order here, because they're not related to "connect". What do you think?

Reply:
I see, yeah you are right it might be like this. I've created for this task `Refactor Wagmi connectors`.